sycl/source/detail/pi.cpp

```diff
@@ -197,7 +197,12 @@ bool trace(TraceLevel Level) {
 
 // Initializes all available Plugins.
 vector_class<plugin> initialize() {
-  vector_class<plugin> Plugins;
+  static bool PluginsInitDone = false;
+  static vector_class<plugin> Plugins;
+  if (PluginsInitDone) {
+    return Plugins;
+  }
```

Comment:
Perhaps
```suggestion
  if (!Plugins.empty())
    return Plugins;
```
could be enough with no need for `PluginsInitDone` variable?

Reply:
I thought that in theory there can be no plugins available and then empty vector cannot mean initialization is needed.

Reply:
Is it possible to run only on CPU with the host backend without any plugin?
If so you are right, this is a useful optimization.

Reply:
Yes, I believe it is valid.